src/core/option.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
```

Comment:
This class provides support for optional objects, which is nice to have support for. It is part of the C++17 standard, but for portability reasons I written a custom version (rather than rely on std::experimental::).